### PR TITLE
Pendle Router

### PR DESCRIPTION
**Pull request by JTraversa**
Added the router for Pendle so I can move towards wider testing efforts.

Im fairly sure the implementation is 99% correct though it needs minor confirmation -- The issues will come in while testing and attempting to create the params necessary both within our tests and in prod.

There are a ton of nested structs and random bs that will need to be packed into some bytes, leaving tons of room for mistakes with packing.

### CODE REVIEW

File: src/adapters/PendleAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import {IPendle} from "../interfaces/IPendle.sol";
+import {IPendleSYToken} from "../interfaces/IPendleSYToken.sol";
+import {IPendleToken}   from "../interfaces/IPendleToken.sol";
+import {IMarketPlace} from "../interfaces/IMarketPlace.sol";
+import {ILender} from "../interfaces/ILender.sol";
+
+import {Pendle} from "../lib/Pendle.sol";
+import {Safe} from "../lib/Safe.sol";
+
+contract YieldAdapter is IAdapter {
```
Comment (simonovic86) [inline src/adapters/PendleAdapter.sol:16]: `PendleAdapter`? 🙂

Comment (alexanderwende) [inline src/adapters/PendleAdapter.sol:16]: We should probably not call it `YieldAdapter`... 😅

**Review by alexanderwende — APPROVED** [review]

File: src/adapters/PendleAdapter.sol [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    function maturity(address pt) public view returns (uint256) {
+        return IPendleToken(pt).expiry();
+    }
+
+    // @notice lendABI "returns" the arguments required in the bytes `d` for the lend function
+    // @returns underlying_ The address of the underlying token
+    // @returns maturity The maturity of the underlying token
+    // @returns minimum The minimum amount of the PTs to receive when spending (amount - fee)
+    // @returns pool The address of the pool to lend to (buy PTs from)
+    function lendABI(
```
[the remark this answers is not in the document]
Comment (JTraversa) [reply src/adapters/PendleAdapter.sol:47]: 👍

File: src/adapters/PendleAdapter.sol (hunk 1 above)
Comment (JTraversa) [reply src/adapters/PendleAdapter.sol]: 👍

**Pull request merged**